File: DivisbleSumPairs.java
// https://www.hackerrank.com/challenges/divisible-sum-pairs/problem?isFullScreen=true

import java.util.List;

public class DivisbleSumPairs {
    public static int divisibleSumPairs(int n, int k, List<Integer> ar) {
        int sum = 0;
        int count = 0;
        for(int i = 0; i < n; i++){
            for(int j = i + 1; j < n; j++){
                sum = ar.get(i) + ar.get(j);
                if(sum % k == 0){
                    count++;
                }
            }
        }
        return count;
    }

    public static void main(String[] args) {
        List<Integer> ar = List.of(1, 2, 3, 4, 5, 6);
        int n = ar.size();
        int k = 5;
        int result = divisibleSumPairs(n, k, ar);
        System.out.println("Number of divisible sum pairs: " + result);
    }
}

File: Python/DivisbleSumPairs.py
# https://www.hackerrank.com/challenges/divisible-sum-pairs/problem?isFullScreen=true

class solution:
    def divisbleSumPairs(n, k, ar):
        sum = 0
        count = 0
        
        for i in range(0, n):
            for j in range(i + 1, n):
                sum = ar[i] + ar[j]
                if sum % k == 0:
                    count += 1
        
        return count
    
if __name__ == "__main__":
    n = 6
    k = 3
    ar = [1, 3, 2, 6, 1, 2]
    result = solution.divisbleSumPairs(n, k, ar)
    print("Number of divisible sum pairs:", result)
    
        